src/ir_Argo.h

```diff
@@ -20,35 +20,53 @@
 
 //  ARGO Ulisse DCI
 
-/*
-	Protocol Description:
-  All in LSB first as it is sent. argo message array will be stored MSB first!
-  do LSB-MSB conversion in sendData
-  Byte 0: const 0	0	1	1	0	1	0	1
-  Byte 1: const 1	0	1	0	1	1	1	1
-  Byte 2: 0 0 0, 3bit Cool/Heat Mode, 2bit start SetTemp LSB first
-  Byte 3: 3bit End SetTemp, 2bit Fan Mode, 3bit RoomTemp LSB first
-  Byte 4: 2bit RoomTemp, 3bit Flap Mode, 3bit OnTimer
-  Byte 5: 8bit OnTimer
-  Byte 6: 8Bit OffTimer
-  Byte 7: 3bit OffTimer, 5bit Time
-  Byte 8: 6bit Time, 1bit Timer On/Off, 1bit Timer Program
-  Byte 9: 1bit Timer Program, 1bit Timer 1h, 1 bit Night Mode, 1bit Max Mode, 1bit Filter, 1bit on/off, 1bit const 0, 1bit iFeel
-  Byte 10: 2bit const 0 1, 6bit Checksum
-  Byte 11: 2bit Checksum
-*/
+union ArgoProtocol {
+  uint8_t raw[kArgoStateLength];  ///< The state in native IR code form
+  struct {
+    // Byte 0
+    uint64_t          :8;  // Typically 0b00110101
+    // Byte 1
+    uint64_t          :8;  // Typically 0b10101111
+    // Byte 2~4
+    uint64_t          :3;
+    uint64_t Mode     :3;
+    uint64_t Temp     :5;  // straddle byte 2 and 3
+    uint64_t Fan      :2;
+    uint64_t RoomTemp :5;  // straddle byte 3 and 4
+    uint64_t Flap     :3;  // SwingV
+    uint64_t          :3;  // OnTimer, maybe hours
+    // Byte 5
+    uint64_t          :8;  // OnTimer, maybe minutes
+    // Byte 6
+    uint64_t          :8;  // OffTimer, maybe minutes
+    // Byte 7
+    uint64_t          :3;  // OffTimer, maybe hours
+    uint64_t          :5;  // Time
+    // Byte 8
+    uint8_t           :6;  // Time
+    uint8_t           :1;  // Timer On/Off
+    uint8_t           :1;  // Timer Program
+    // Byte 9
+    uint8_t           :1;  // Timer Program
+    uint8_t           :1;  // Timer 1h
+    uint8_t Night     :1;
+    uint8_t Max       :1;
+    uint8_t           :1;  // Filter
+    uint8_t Power     :1;
+    uint8_t           :1;  // const 0
+    uint8_t iFeel     :1;
+    // Byte 10~11
+    uint16_t          :2;  // const 01
+    uint16_t Sum      :8;
+    uint16_t          :6;
```

Comment:
Discussion only: Per your comment/advice about alignment, it may be better to make all of this (`Byte[8-11]`) an `uint32_t`?
No idea if it produces better/more efficient code or not.

It might be safer to prevent accidental alignment issues/changes in the future if we just use a series of the biggest integer sizes support for a given "state" for this/other A/Cs?

Thoughts?

I'll defer to your wisdom on the matter as you're more of an expert on this topic than I am.

Reply:
I need to check some information

Reply:
What you said makes sense. If there is no named field straddling bytes, we should use uint8_t everywhere, since it is the smallest integer size, we won't meet the alignment issues; Else we just use a series of the biggest integer sizes support :-)